content/client-lib-development-guide/features.textile

```diff
@@ -1534,40 +1478,59 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + getDeviceInfo() => io DeviceDetails
+  id: String
+  platform: DeviceType
+  type: DeviceType
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
```

Comment:
I think this requires more thought. For launch I think we should consider the `updateToken` or `localDeviceToken` being generated along with the device ID, and updatable with `push-admin` rights. If we want a robust way for the device to be able to prove that it is what it claims then we could have an asymmetric keypair generated locally on the device and used by the device to sign token requests or whatever. So I don't think we should be putting things in the IDL that we don't intend to implement and release without being more confident that they will be the right thing when we come to implement them.